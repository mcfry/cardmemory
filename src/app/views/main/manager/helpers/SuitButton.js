import React from "react";
import classNames from 'classnames';

class SuitButton extends React.PureComponent {
	render() {
		const {active, suit, suitClick, blackImg, whiteImg} = this.props;

		return (
			<button className={classNames('list-group-item', 'list-group-item-action', { active: active})} onClick={suitClick}>
			  	{!active ? (
			    	<img className="card-nav-suit" alt="hearts" src={blackImg}/>
			    ) : (
			    	<img className="card-nav-suit" alt="hearts" src={whiteImg}/>
				)}
			    &nbsp;<span>{suit.charAt(0).toUpperCase() + suit.substr(1)}</span>
		  	</button>
		);
	}
}

export default SuitButton;