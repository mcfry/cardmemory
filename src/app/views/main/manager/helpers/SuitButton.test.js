import React from 'react';
import renderer from 'react-test-renderer';
import SuitButton from './SuitButton';

test('Changes when active', () => {
	const comp = renderer.create(
		<SuitButton active={true} suit={'hearts'}/>
	);

	let tree = comp.toJSON();
	expect(tree).toMatchSnapshot();

	tree.props.active = false;

	tree = comp.toJSON();
	expect(tree).toMatchSnapshot();
});