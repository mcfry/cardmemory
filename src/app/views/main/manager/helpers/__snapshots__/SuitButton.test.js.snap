// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changes when active 1`] = `
<button
  className="list-group-item list-group-item-action active"
>
  <img
    alt="hearts"
    className="card-nav-suit"
  />
   
  <span>
    Hearts
  </span>
</button>
`;

exports[`Changes when active 2`] = `
<button
  className="list-group-item list-group-item-action active"
>
  <img
    alt="hearts"
    className="card-nav-suit"
  />
   
  <span>
    Hearts
  </span>
</button>
`;
